File: sort_the_odd.rb
def sort_array(source_array)
  sorted_odds = []
  source_array.each { |i| sorted_odds << i if i % 2 != 0}
  sorted_odds.sort!
  source_array.each_with_index { |num, i| source_array[i] = sorted_odds.shift if num % 2 != 0 }
  source_array
end

puts sort_array([5, 3, 2, 8, 1, 4])

File: sort_the_odd.js
function sortArray(array) {
  var sortedOdds = array.filter(n => (n % 2 !== 0)).sort((a, b) => a - b)
  array.forEach((num, index) => {
    if (num % 2 !== 0) {
      array[index] = sortedOdds.splice(0,1)[0];
    }
  })
  return array;
}

console.log(sortArray([21,7,34,2,9,6,45,3]));
